src/common/AndroidUtils.ts

```diff
@@ -471,49 +430,52 @@ export class AndroidSDKUtils {
         emulatorPort: number,
         serverAddress: string | undefined,
         serverPort: string | undefined
-    ): Promise<boolean> {
-        try {
-            if (appBundlePath && appBundlePath.trim().length > 0) {
-                AndroidSDKUtils.logger.info(
-                    `Installing app ${appBundlePath.trim()} to emulator`
-                );
-                const pathQuote = process.platform === WINDOWS_OS ? '"' : "'";
-                const installCommand = `${AndroidSDKUtils.getAdbShellCommand()} -s emulator-${emulatorPort} install -r -t ${pathQuote}${appBundlePath.trim()}${pathQuote}`;
-                CommonUtils.executeCommandSync(installCommand);
-            }
+    ): Promise<void> {
+        let thePromise: Promise<{ stdout: string; stderr: string }>;
+        if (appBundlePath && appBundlePath.trim().length > 0) {
+            AndroidSDKUtils.logger.info(
+                `Installing app ${appBundlePath.trim()} to emulator`
+            );
+            const pathQuote = process.platform === WINDOWS_OS ? '"' : "'";
+            const installCommand = `${AndroidSDKUtils.getAdbShellCommand()} -s emulator-${emulatorPort} install -r -t ${pathQuote}${appBundlePath.trim()}${pathQuote}`;
+            thePromise = CommonUtils.executeCommandAsync(installCommand);
+        } else {
+            thePromise = Promise.resolve({ stdout: '', stderr: '' });
+        }
 
-            let launchArgs =
-                `--es "${PreviewUtils.COMPONENT_NAME_ARG_PREFIX}" "${compName}"` +
-                ` --es "${PreviewUtils.PROJECT_DIR_ARG_PREFIX}" "${projectDir}"`;
+        return thePromise
+            .then(() => {
```

Comment:
This feels weird. I know why it is the way it is, and I don't know of a better solution. But it still feels weird.

Reply:
yeah, I tried to come up with something better but couldn't think of something.